lib/streamlit/web/server/browser_websocket_handler.py

```diff
@@ -171,6 +186,20 @@ def open(self, *args: Any, **kwargs: Any) -> Awaitable[None] | None:
                 if self._validate_xsrf_token(csrf_protocol_value):
                     user_info.update(self._parse_user_cookie(raw_cookie_value))
 
+                    # Also read in tokens if token cookie exists
+                    raw_token_cookie_value = self.get_signed_cookie(TOKENS_COOKIE_NAME)
+                    if raw_token_cookie_value:
+                        all_tokens = json.loads(raw_token_cookie_value)
+
+                        # Filter tokens based on expose_tokens configuration
+                        filtered_tokens = {}
+                        for token_type in self.expose_tokens:
+                            token_key = f"{token_type}_token"
+                            if token_key in all_tokens:
+                                filtered_tokens[token_type] = all_tokens[token_key]
+
+                        user_info["tokens"] = filtered_tokens
```

Comment:
Would this be a good option to put `TokensProxy(filtered_tokens)`

Reply:
Possibly? But by the same logic, would it not make sense to create the UserInfo object right then and there?

I'm operating by analogy (as you do in an unfamiliar codebase) and I was assuming there was a reason. For instance, the user_info object might need robust serialization which standard dict is better suited for than classes etc. 

It feels safer to leave it as it is, but I can change it if you want me to.

Reply:
Ah that makes sense. I might have a different approach to make things cleaner.